File: qleany/common/direct_access/common/persistence-registration.py


from qleany.common.direct_access.common.repository.repository_factory import RepositoryFactory


def register_persistence() -> RepositoryFactory:
    factory = RepositoryFactory()

    # entity
    from qleany.common.direct_access.entity.entity_db_table_group import EntityDbTableGroup
    from qleany.common.direct_access.entity.entity_repository import EntityRepository
    factory.register(EntityRepository, EntityDbTableGroup)



    return factory

File: crates/common/src/direct_access/persistence_registration.rs
use crate::direct_access::repository_factory::RepositoryFactory;
use crate::database::sqlite_db_context::SqliteDbContext;

pub fn register() -> RepositoryFactory {
    let db_context = Box::new(SqliteDbContext::new());
    let factory = RepositoryFactory::new();

    factory
}
